### PR TITLE
Add fullscreen workspace selector

### PR DESCRIPTION
**Pull request by thejch**
#### Describe your PR, what does it fix/add?
Add a workspace selector for fullscreen states. This allows more control for fullscreen workspaces, e.g. specifying the gaps when maximized etc.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
not really

#### Is it ready for merging, or does it need work?
ready

### CODE REVIEW

File: src/desktop/Workspace.cpp [hunk 1]
```diff
@@ -442,6 +444,37 @@ bool CWorkspace::matchesStaticSelector(const std::string& selector_) {
...
             }
 
+            if (cur == 'f') {
+                if (!prop.starts_with("f[") || !prop.ends_with("]")) {
+                    Debug::log(LOG, "Invalid selector {}", selector);
+                    return false;
+                }
+
+                prop               = prop.substr(2, prop.length() - 3);
+                const auto FSSTATE = configStringToInt(prop);
```
Comment (vaxerski) [inline src/desktop/Workspace.cpp:454]: I'd rather use `std::stoi` here, as configStringToInt returns 0 on error

Just remember to wrap in a try catch

Comment (thejch) [reply src/desktop/Workspace.cpp]: fixed

**Review by vaxerski — APPROVED** [review]
lgtm

**Comment by vaxerski** [comment]
wiki mr neededd

**Pull request merged**